src/main/java/mate/academy/internetshop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.internetshop.dao.impl;
+
+import java.util.Optional;
+import java.util.stream.IntStream;
+import mate.academy.internetshop.dao.ShoppingCartDao;
+import mate.academy.internetshop.db.Storage;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.ShoppingCart;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return Storage.addShoppingCard(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        int index = IntStream.range(0, Storage.shoppingCarts.size())
+                .filter(i -> Storage.shoppingCarts.get(i).getId().equals(shoppingCart.getId()))
+                .findFirst().orElse(-1);
+        if (index == -1) {
+            return create(shoppingCart);
```

Comment:
We don't need to create a shopping cart if it does not exist. At least, the update method is not the right place for it. Move to service, please

Reply:
Removed to shopingCartService.addProduct.